assets/js/wfparameters/RenameEntries.js

```diff
@@ -0,0 +1,194 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import WorkBenchAPI from '../WorkbenchAPI'
+
+var api = WorkBenchAPI();
+export function mockAPI(mock_api) {
+    api = mock_api;
+}
+
+export class RenameEntry extends React.Component {
+    static propTypes = {
+        colname: PropTypes.string.isRequired,
+        newColname: PropTypes.string.isRequired,
+        onColRename: PropTypes.func.isRequired,
+        onEntryDelete: PropTypes.func.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            inputValue: this.props.newColname
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleKeyPress = this.handleKeyPress.bind(this);
+        this.handleBlur = this.handleBlur.bind(this);
+        this.handleFocus = this.handleFocus.bind(this);
+        this.handleDelete = this.handleDelete.bind(this);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if(nextProps.newColname != this.state.inputValue) {
+            this.setState({inputValue: nextProps.newColname});
+        }
+    }
+
+    handleChange(event) {
+        //this.props.onColRename(this.props.colname, event.target.value);
+        this.setState({inputValue: event.target.value});
+    }
+
+    handleBlur() {
+        if(this.state.inputValue != this.props.newColname) {
+            this.props.onColRename(this.props.colname, this.state.inputValue);
+        }
+    }
+
+    handleKeyPress(event) {
+        if((event.key == 'Enter') && (this.state.inputValue != this.props.newColname)) {
+            this.props.onColRename(this.props.colname, this.state.inputValue);
+        }
+    }
+
+    handleFocus(event) {
+        event.target.select();
+    }
+
+    handleDelete() {
+        this.props.onEntryDelete(this.props.colname);
+    }
+
+    render() {
+        // The class names below are used in testing.
+        // Changing them would require updating the tests accordingly.
+        return (
+            <div>
+                <div className={'rename-column'} style={{width: '40%', float: 'left'}}>{this.props.colname}</div>
```

Comment:
Do not use `style`. Use CSS.

Passing object parameters like this forces DOM writes with every render, even when nothing changes.